corehq/apps/geospatial/static/geospatial/js/models.js

```diff
@@ -591,12 +628,12 @@ hqDefine('geospatial/js/models', [
 
         self.clearActivePolygon = function () {
             if (self.activeSavedPolygon) {
-                // self.selectedSavedPolygonId('');
-                self.removePolygonsFromFilterList(self.activeSavedPolygon.geoJson.features);
+                self.selectedSavedPolygonId('');
                 removeActivePolygonLayer();
                 self.activeSavedPolygon = null;
                 self.btnSaveDisabled(false);
                 self.btnExportDisabled(true);
+                updateSelectedSavedPolygonParam();
```

Comment:
is it possible to instead subscribe to change to `selectedSavedPolygonId` to call this function?

Reply:
Definitely. I see we are already calling `updateSelectedSavedPolygonParam()` in the subscribe function for `selectedSavedPolygonId` so can remove it from here ([7f5da66](https://github.com/dimagi/commcare-hq/pull/34511/commits/7f5da662bf8de6c0c5901313594e2291ff0a9d2d)).